spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/sql/TestNamespaceSQL.java

```diff
@@ -134,15 +134,9 @@ public void testListNamespace() {
 
     List<Object[]> namespaces = sql("SHOW NAMESPACES IN %s", catalogName);
 
-    if (isHadoopCatalog) {
-      Assert.assertEquals("Should have 1 namespace", 1, namespaces.size());
-      Set<String> namespaceNames = namespaces.stream().map(arr -> arr[0].toString()).collect(Collectors.toSet());
-      Assert.assertEquals("Should have only db namespace", ImmutableSet.of("db"), namespaceNames);
-    } else {
-      Assert.assertEquals("Should have 2 namespaces", 2, namespaces.size());
-      Set<String> namespaceNames = namespaces.stream().map(arr -> arr[0].toString()).collect(Collectors.toSet());
-      Assert.assertEquals("Should have default and db namespaces", ImmutableSet.of("default", "db"), namespaceNames);
-    }
+    Assert.assertEquals("Should have 2 namespaces", 2, namespaces.size());
+    Set<String> namespaceNames = namespaces.stream().map(arr -> arr[0].toString()).collect(Collectors.toSet());
+    Assert.assertEquals("Should have default and db namespaces", ImmutableSet.of("default", "db"), namespaceNames);
```

Comment:
What changed this behavior?

Reply:
The assertion checks if the namespace that `sql("CREATE NAMESPACE %s", fullNamespace);` creats is in fact created or not.

Now, instead of having only one namespace i.e. "db" which is created using `CREATE NAMESPACE` stmt we have two; one is "db" and the other is "default" that is created in the constructor of the `SparkTestBaseWithCatalog` class during setup.

Assuming the objective of the assertion was to check if the "db" namespace was created or not and even after the changes that this PR introduces the aim is still met, thus there's no change in behavior.

Reply:
The question is whether the test aims to test the creation using the `CREATE NAMESPACE IF NOT EXISTS default` sql too, or the only goal was to test if the NS is created by using the catalog API.

Someone with more knowledge about the Spark test could be good here.